api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/merge/Dot.kt

```diff
@@ -0,0 +1,126 @@
+package org.jetbrains.kotlinx.dl.api.core.layer.merge
+
+import org.tensorflow.Operand
+import org.tensorflow.op.Ops
+import org.tensorflow.op.Scope
+import org.tensorflow.op.core.*
+import org.tensorflow.op.linalg.MatMul
+import org.tensorflow.op.linalg.Transpose
+import org.tensorflow.op.math.Mul
+import org.tensorflow.op.math.Rsqrt
+import org.tensorflow.op.math.Square
+
+public class Dot(
+    val axis: IntArray,
+    val normalize: Boolean = false,
+    name: String = ""
+) : AbstractMerge("Dot", name) {
+
+    constructor(axis: Int) : this(axis = IntArray(2) { axis })
+
+    private fun l2Normalize(scope: Scope?, x: Operand<Float>, axis: IntArray?): Operand<Float> {
+        val squareSum: Operand<Float> = ReduceSum.create(
+            scope,
+            Square.create(scope, x),
+            Constant.create(scope, axis),
+            ReduceSum.keepDims(true)
+        )
+        val invNorm: Operand<Float> = Rsqrt.create(
+            scope,
+            org.tensorflow.op.math.Maximum.create(
+                scope, squareSum, Constant.create(scope, 1e-12f)
+            )
+        )
+        return Mul.create(scope, x, invNorm)
+    }
+
+    private fun batchDot(scope: Scope?, x: Operand<Float>, y: Operand<Float>, axis: IntArray): Operand<Float> {
```

Comment:
Probably this function could be moved out of class, made public and KDocs could be useful too